rc/tools/make.kak

```diff
@@ -1,7 +1,10 @@
 declare-option -docstring "shell command run to build the project" \
     str makecmd make
-declare-option -docstring "pattern that describes lines containing information about errors in the output of the `makecmd` command" \
+declare-option -hidden -docstring "pattern that find lines containing information about errors in the output of the `makecmd` command" \
     str make_error_pattern " (?:fatal )?error:"
+declare-option -docstring "pattern that describes lines containing information about errors in the output of the `makecmd` command. Capture groups must be: 1: filename 2: line number 3: optional column 4: optional error description" \
+    regex make_error_line_pattern "^(?:\w:)?([^:\n]+):(\d+):(?:(\d+):)? (?:fatal )?error:([^\n]+)?"
```

Comment:
I think we could probably just change make_error pattern here, I figure you did it this way to avoid breaking user configuration but I am not sure this option is used much. I also wonder if we could use named captures here to support different ordering of capture groups, this would require some extra C++ side feature to be usable though,

Reply:
changed the code to reuse the old option now. 
Regarding named capture group, I leave it unchanged because I don't know how to refer named capture group, do we support it in the kak file right now?

Reply:
No, this would need some extra work to expose capture name -> capture id from the regex. It would be a nice feature but I am not quite clear on the design yet.